modules/social_features/social_event/modules/social_event_an_enroll/src/Form/EventAnEnrollActionForm.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Drupal\social_event_an_enroll\Form;
+
+use Drupal\social_event\Form\EnrollActionForm;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\node\Entity\Node;
+
+/**
+ * Class EventAnEnrollActionForm.
+ *
+ * @package Drupal\social_event_an_enroll\Form
+ */
+class EventAnEnrollActionForm extends EnrollActionForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'event_an_enroll_cancel_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $nid = $this->routeMatch->getRawParameter('node');
```

Comment:
We should add this as a form argument:
```
public function buildForm(array $form, FormStateInterface $form_state, NodeInterface $node) {
```

Then when calling the form we should do:
```
getForm('Drupal\social_event_an_enroll\Form\EventAnEnrollActionForm', $node);
```
as per https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21FormBuilder.php/function/FormBuilder%3A%3AgetForm/8.5.x

Reply:
This should also be fixed for the `EnrollActionForm` class or a follow-up should be created.

Reply:
It was fixed here, but I see that `EnrollActionForm` doesn't use `routeMatch`, so I haven't change anything there.